### PR TITLE
added docs

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
Currently, it looks somewhat like this:

I think it looks fairly good.

Screenshot 2023-04-03 at 17 27 40

Screenshot 2023-04-03 at 17 27 51

### CODE REVIEW

File: {{cookiecutter.project_name}}/pyproject.toml [hunk 1]
```diff
@@ -31,11 +31,11 @@ name = "{{cookiecutter.license}}"
 [project.optional-dependencies]
 dev = [
   "cruft",
-  "mypy",
+  "mypy",  
+  "pandas-stubs", # mypy stubs for pandas
   "pre-commit==2.20.0,<2.21.0",
   "ruff==0.0.254", # important that these match the pre-commit hooks
   "black[jupyter]==22.8.0", # important that these match the pre-commit hooks
```
Comment (MartinBernstorff) [inline {{cookiecutter.project_name}}/pyproject.toml:38]: This re-ordering will cause merge conflicts for all downstream projects, so would prefer to omit it :-)

File: {{cookiecutter.project_name}}/.github/recommended_repo_setup.md
Comment (MartinBernstorff) [inline {{cookiecutter.project_name}}/.github/recommended_repo_setup.md:1]: This file was tremendously helpful to have in the newly created repo during setup, so I don't have to navigate to the `swift-python-cookiecutter` readme during setup. Would strongly prefer keeping it, and then perhaps pointing to it in the `swift-python-cookiecutter` readme.

**Review by MartinBernstorff — CHANGES_REQUESTED** [review]

**Comment by MartinBernstorff** [comment]
Looks sexy, I'm game! 

Might want to add a `docs` action to `invoke` as well? Maybe both a `docs-build` and `docs-view`?

File: {{cookiecutter.project_name}}/.github/recommended_repo_setup.md
Comment (KennethEnevoldsen) [reply {{cookiecutter.project_name}}/.github/recommended_repo_setup.md:1]: Hmm would probably prefer having it here (having all the information to set up the repo in once place), but I am not that opinionated on this. Def. don't want to maintain docs in two places.

**Comment by MartinBernstorff** [comment]
> Good point! Added it as one command with the interface:
> 
> ```
> inv docs # build and view
> inv docs -b # only build
> inv docs -v # only view
> ```

Looks great!

I think I'd prefer it for `inv docs` to only build, and then have the `-v` flag to view. I think this matches the typical makefile better.

Couldn't find many examples, but e.g. Dan on scandeval: https://github.com/saattrupdan/ScandEval/blob/main/makefile

File: README.md [hunk 2]
```diff
@@ -30,42 +30,10 @@ Where `repository url` is a link to an empty repository.
...
 ## Recommened setup for the repository
-### Github personal access token
-For pre-commit actions to be able to commit changes to a PR, or for the Cruft update to be able to create a PR with updates from the upstream repo, you need a personal access token (PAT) with the following settings:
 
-**Expiration**: No expiration
-**scopes:**
-* `repo`
-* `workflow`
+To see the recommended setup for the repository, see the following file:
+`{{cookiecutter.project_name}}/.github/recommended_repo_setup.md`
```
Comment (MartinBernstorff) [inline README.md:35]: Do we want this in the readme? That's two lines I'd have to delete every time.

Depends on whether we're targeting only internal use, or also for e.g. potential students. If it's only for us three, I'd remove it. What do you think? :-)

File: {{cookiecutter.project_name}}/tasks.py [hunk 3]
```diff
@@ -271,3 +275,20 @@ def pr(c: Context):
...
+    if not build and not view:
+        build = True
+        view = True
+
+    if build:
+        echo_header(f"{Emo.DO} Building docs")
+        c.run("sphinx-build -b html docs docs/_build/html")
+    if view:
+        echo_header(f"{Emo.EXAMINE} open docvs in browser")
+        c.run("open docs/_build/html/index.html")
```
Comment (MartinBernstorff) [inline {{cookiecutter.project_name}}/tasks.py:294]: This command is not cross-compatible with Windows. 

Not an issue with the docs command, but I think we'd like WIndows compatability for the other commands :-)

File: {{cookiecutter.project_name}}/tasks.py [hunk 4]
```diff
@@ -271,3 +275,20 @@ def pr(c: Context):
     test(c)
     update_branch(c)
     update_pr(c)
+
+
+@task
+def docs(c: Context, build: bool=False, view: bool = False):
```
Comment (MartinBernstorff) [inline {{cookiecutter.project_name}}/tasks.py:281]:
```suggestion
def docs(c: Context, build: bool = False, view: bool = False):
```

**Review by MartinBernstorff — APPROVED** [review]
Only minor comments!

File: {{cookiecutter.project_name}}/tasks.py (hunk 3 above)
Comment (KennethEnevoldsen) [reply {{cookiecutter.project_name}}/tasks.py:305]: It should now be fixed

File: README.md (hunk 2 above)
Comment (KennethEnevoldsen) [reply README.md:35]: I assume you thought it was the created one?

**Comment by KennethEnevoldsen** [comment]
> I think I'd prefer it for inv docs to only build, and then have the -v flag to view. I think this matches the typical makefile better.
> 
> Couldn't find many examples, but e.g. Dan on scandeval: https://github.com/saattrupdan/ScandEval/blob/main/makefile

Hmm, but what if I just want to view it?

Alternatively it could be (which was my initial idea):
```
build-docs # build
build-docs  -v # build docs and view
view-docs # just view
```

I decided against it though as it wasn't as clean

**Comment by KennethEnevoldsen** [comment]
I initially did the following:

```
build-docs # build
build-docs -v # build and view
view-docs # just view
```
but I don't think it was as clean so now I did:
```
docs # build
docs -v # build and view
docs --view-only # or -i (which is apparently the assigned flag). 
```

File: README.md (hunk 2 above)
Comment (MartinBernstorff) [reply README.md:35]: Ahh, yes, thanks for clarifying!

**Comment by MartinBernstorff** [comment]
Ah, I see. You have more experience around docs workflows, so whichever you prefer 👍

**Pull request merged**